### ISSUE TITLE
[Bug] When evaluate_during_training=True, after the evaluation and before the next epoch, the model is in eval mode

### ISSUE THREAD

**Issue by nilboy** [ThilinaRajapakse/simpletransformers#941]
**Describe the bug**
When evaluate_during_training=True, after the evaluation and before the next epoch, the model is in eval mode. That way, batch-norm or dropout layers will work in eval mode instead of training mode.

https://github.com/ThilinaRajapakse/simpletransformers/blob/master/simpletransformers/language_modeling/language_modeling_model.py#L826

This is my training loss curve.

![屏幕快照 2021-01-08 下午4 54 10](https://user-images.githubusercontent.com/17962699/103994615-32ab4800-51d2-11eb-8993-0b6c7a8de964.png)
